### PR TITLE
BUILD - raised state for todos and added date picker value. Created f…

### PR DESCRIPTION
**Pull request by Joshibbotson**
-Added date picker so todos have date values.
- Added date-fns library to utilise isToday() and isThisWeek() functions for filtering todo tasks.
- Added simple nav buttons linked to useEffect to filter todos and update DOM accordingly.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,13 +1,92 @@
...
+import { parseISO, isToday, isThisWeek } from "date-fns"
 
 function App() {
-  return (
-    <>
-      <h1 className="text-3xl font-bold underline">Hello world!</h1>
-      <TodoForm />
-    </>
-  );
+    const [inputValue, setInputValue] = useState("")
```
Comment (Gw3i) [inline src/App.jsx]: Why is the todo form now in the App() when there is a TodoForm Component? The logic of adding todos should be in the TodoForm Component.
The App Component is just for displaying the components, no business logic or very little

File: src/App.jsx [hunk 2]
```diff
@@ -1,13 +1,92 @@
...
+    const [dateValue, setDateValue] = useState(null)
+    const [todos, setTodos] = useState([
+        { id: nanoid(), name: "Buy coffee" },
+        { id: nanoid(), name: "Water plants" },
+    ])
+    const [filteredTodos, setFilteredTodos] = useState([])
```
Comment (Gw3i) [inline src/App.jsx]: I think a TodoList Component would be good to display all todos in a list. There you could use the filters, but not in the App()

File: src/App.jsx [hunk 3]
```diff
@@ -1,13 +1,92 @@
...
+                    <button
+                        onClick={() => {
+                            setCurrentFilter("thisWeek")
+                        }}
+                    >
+                        This Week
+                    </button>
+                </nav>
+            </header>
+            <TodoForm
```
Comment (Gw3i) [inline src/App.jsx]: There is no need to pass these props into TodoForm. And best practice tip is not to pass setter functions as props :)

File: src/Components/TodoForm.jsx [hunk 4]
```diff
@@ -1,55 +1,57 @@
...
+import { useEffect, useState } from "react"
 
-export default function TodoForm() {
-  const [inputValue, setInputValue] = useState("");
-  const [todos, setTodos] = useState([
-    { id: nanoid(), name: "Buy coffee" },
-    { id: nanoid(), name: "Water plants" },
-  ]);
+export default function TodoForm({
+    todos,
```
Comment (Gw3i) [inline src/Components/TodoForm.jsx]: No need for props right now

**Review by Gw3i — CHANGES_REQUESTED** [review]

**Comment by Joshibbotson** [comment]
Too many conflicts with new commits.

**Pull request closed without merging**